src/main/java/org/skriptlang/skript/bukkit/recipes/elements/SecRegisterRecipe.java

```diff
@@ -0,0 +1,272 @@
+package org.skriptlang.skript.bukkit.recipes.elements;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.config.Node;
+import ch.njol.skript.config.SectionNode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.*;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.NamespacedUtils;
+import ch.njol.skript.util.RecipeUtils.RegisterRecipeEvent;
+import ch.njol.skript.util.RecipeUtils.RecipeType;
+import ch.njol.skript.util.RecipeUtils.RegisterRecipeEvent.*;
+import ch.njol.skript.util.RecipeUtils.RegisterRecipeEvent.CraftingRecipeEvent.*;
+import ch.njol.skript.variables.Variables;
+import ch.njol.util.Kleenean;
+import org.bukkit.Bukkit;
+import org.bukkit.NamespacedKey;
+import org.bukkit.event.Event;
+import org.bukkit.inventory.*;
+import org.bukkit.inventory.recipe.CookingBookCategory;
+import org.bukkit.inventory.recipe.CraftingBookCategory;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+@Name("Register Recipe")
+@Description({
+	"Create a custom recipe for any of the following types:",
+	"Shaped, Shapeless, Blasting, Furnace, Campfire, Smoking, Smithing Transform, Smithing Trim or Stonecutting",
+	"NOTES:",
+	"All recipes except Smithing Trim require a 'result item'",
+	"Shaped and Shapeless Ingredients allows custom items only on Paper",
```

Comment:
since when? Bukkit has always allowed custom ingredients in recipes.

Reply:
Yeah that's my fault. I had made that description before looking into RecipeChoice

Reply:
Ahh ok, thats fair. Paper has their own ItemStack option dont they?!?!?! 
RecipeChoice can be a smidge confusing.

Reply:
Yeah Paper does have an ItemStack constructor, but Fuse talked me into RecipeChoice as it allows to use `all wood` as 1 ingredient